std/typecons.d

```diff
@@ -2306,10 +2306,15 @@ Returns:
         {
             return _value is nullValue;
         }
-        else
+
+        //Need to use 'is' if T is a float type
+        //because NaN != NaN
+        static if (isFloatingPoint(T))
         {
-            return _value == nullValue;
+            return _value is nullValue;
         }
```

Comment:
Please add an `else` branch here, because this might otherwise trigger a warning about unreachable code.